From 6ad1de37d9667a3986e428b021c5b8212a1d061a Mon Sep 17 00:00:00 2001
From: Yaniv Levinsky <ylevinsky@habana.ai>
Date: Mon, 10 May 2021 15:10:56 +0300
Subject: [PATCH] set greco-asic-install target to copy images to vm

Add to your .bashrc:

export HL_INSTALL_HOST_IP=<host_ip>
export HL_INSTALL_LAB_IP=<host_ip>
export HL_ROOT_PASS=Hab12345

Change-Id: I0ee569c9dd5f096ff01747d29194f425c58acbf0
---

diff --git a/scripts/make/common/Makefile b/scripts/make/common/Makefile
index f09e01c..58ffcf3 100644
--- a/scripts/make/common/Makefile
+++ b/scripts/make/common/Makefile
@@ -188,7 +188,7 @@
 SSH := ssh $(SSHOPTS)
 DEV_NAME := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 
-.PHONY: hl-install-host hl-install-lab
+.PHONY: hl-install-host hl-install-lab hl-install-firmware
 hl-install-host:
 	sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/$(DEV_NAME)-fit.itb				root@$(HL_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(DEV_NAME)/
 	sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/$(DEV_NAME)-boot-fit.itb			root@$(HL_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(DEV_NAME)/
@@ -199,6 +199,11 @@
 	sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin			root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	sshpass -p$(HL_PASSWD) $(SCP) $(UBOOT_BASE)/u-boot					root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	sshpass -p$(HL_PASSWD) $(SCP) $(LINUX_BASE)/vmlinux					root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+hl-install-firmware:
+	sshpass -p$(HL_PASSWD) $(SSH) labuser@$(HL_INSTALL_HOST_IP) 			'sudo rm -rf ~/temp/$(HL_DEV)/ && mkdir -p ~/temp/$(HL_DEV)/'
+	sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/$(DEV_NAME)-*.itb			labuser@$(HL_INSTALL_HOST_IP):temp/$(DEV_NAME)/
+	sshpass -p$(HL_PASSWD) $(SSH) labuser@$(HL_INSTALL_HOST_IP)				'sudo chown -R root:root ~/temp/$(HL_DEV)/'
+	sshpass -p$(HL_PASSWD) $(SSH) labuser@$(HL_INSTALL_HOST_IP)				'sudo sudo cp ~/temp/$(HL_DEV)/*.itb /lib/firmware/habanalabs/$(DEV_NAME)/'
 
 PLDM_HEX_DIR ?= $(IMAGES_DIR)/tools/hl_shared_dir/pldm
 ${PLDM_HEX_DIR}:
diff --git a/scripts/make/greco/Makefile b/scripts/make/greco/Makefile
index f20b512..3322b96 100644
--- a/scripts/make/greco/Makefile
+++ b/scripts/make/greco/Makefile
@@ -15,7 +15,7 @@
 
 .PHONY: greco-install-host
 greco-install-host:
-	@$(MAKE) hl-install-host HL_INSTALL_HOST_IP=$(HL_INSTALL_HOST_IP)
+	@$(MAKE) hl-install-firmware HL_INSTALL_HOST_IP=$(HL_INSTALL_HOST_IP)
 
 .PHONY: greco-asic-install greco-onix-install
 greco-onix-install: greco_arm-onix-target greco-install-host
